### PR TITLE
HetTree

### PR DESCRIPTION
**Pull request by naderawadala**
Issue with adding elements in a BFS manner, currently the program works like this:
- paramOne becomes the root
- paramTwo becomes root leftChild
- paramThree becomes root rightChild
- paramTwo becomes 'parent'
- paramFour becomes leftChild of paramTwo
- paramThree becomes 'parent'
- paramFive becomes rightChild of paramThree

When it should be:
```

        paramOne 
        /        \
paramTwo    paramThree
   /     \                 
Four Five        
```

trackPlace is a variable that increments on each instatiation, the intent was to have it determine whether the new node would go left or right, depending on whether it was even/odd(with it starting as odd so that it could be in the BFS order of left to right).

### CODE REVIEW

File: src/main/java/com/clouway/javaintro/core/HTree.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        Node newNode = new Node(object);
+        if (root == null) {
+            root = newNode;
+            return;
+        }
+
+        Node focusNode = root;
+        Node parent;
+
+        trackPlace++;
```
Comment (tsbonev) [inline src/main/java/com/clouway/javaintro/core/HTree.java]: This increment doesn't affect the variable.

File: src/main/java/com/clouway/javaintro/core/HTree.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        while (true) {
```
Comment (tsbonev) [inline src/main/java/com/clouway/javaintro/core/HTree.java]: You can abstract this away and rid yourself of this while loop by implementing some methods in the Node class such as a check if it is full (whether any of its two children are null) and a method to add a child, which you would call if it is not full, that can handle adding a node to the left or right depending on which one is free.

Then you can have a function that will loop through the children of your root element and return the first one it finds that is not full.

BFS is implemented with a Queue structure, the idea being that you would add a node to the queue, check if it is full, return it if not or queue its children if it is, then dequeue the next element in the queue and continue on. This way you check all of the children of a node in the order that they are added to the queue.

Comment (naderawadala) [reply src/main/java/com/clouway/javaintro/core/HTree.java]: Should be fixed

**Comment by naderawadala** [comment]
Closing PR in order to create a new PR with a proper name

**Pull request closed without merging**